scripts/diff.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+
+import glob
+import os
+import argparse
+
+
+def get_latest_chunk_ids(dirs):
+    chunks = []
+    for d in dirs:
+        for f in glob.glob(os.path.join(d, "training.*.gz")):
+            chunks.append(int(os.path.basename(f).split('.')[-2]))
+    chunks.sort(reverse=True)
+    return chunks
+
+
+def main(argv):
+    a = get_latest_chunk_ids([argv.input])
+    n = min(argv.wsize, len(a))
+    b = get_latest_chunk_ids(argv.dirs)
+    diff = set(a[:n]) - set(b)
+    for i in diff:
```

Comment:
I'm not sure about python set - does it iterate in sorted order from smallest to largest?

Reply:
Set iteration order is undefined, and in practice on CPython is hash dependent (read: reproducible within a given process, but ~random between processes).